### ISSUE TITLE
11. Recursive Reporters 1 (LA Review - Sp15)

### ISSUE THREAD

**Issue by lmock** [cs10/bjc-r#523]
```
Your Name:

Overview of Lab Comments: i.e. Was this lab good or bad? Too fast? Too Slow? Does it need something new?

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link:
      Type of Issue: i.e. (code) bug, grammar, typo, lack of clarity, etc
      Severity: i.e. "prevents students from continuing", "minor inconvenience", etc
      Comments: Description of the problem, what can be done to fix it, etc.
- [ ] Page Link:
      Type of Issue: i.e. (code) bug, grammar, typo, lack of clarity, etc
      Severity: i.e. "prevents students from continuing", "minor inconvenience", etc
      Comments: Description of the problem, what can be done to fix it, etc.
```

**Comment by mfujimoto** [comment]
Your Name: Mason Fujimoto

Overview of Lab Comments: A very difficult but satisfying lab. I thought the exercise on subsets was especially useful for testing a student's understanding of a function's output. Despite the difficulty and expectant length of the lab, I think that every exercise is valuable and I would not cut down the lab at all. I feel that doing this lab proves and strengthens ones understanding of list functions which are vitally important for many potential student projects.

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-and-efficiency.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        
        Type of Issue: gives student answer

      Severity: noticable (may result in students not working through and understanding the problem)

      Comments: The page preceding this one gives hints to create the subsets block in the form of a quiz. When the student presses next page, they will immediately see the answer instead of being told to start their attempt of the block themselves. I think that even if the answer is included in the lab, students should have attempted to solve the problem before seeing it. A message to start creating the block on the previous page would rectify this issue

**Comment by anushasyed** [comment]
Anusha Syed 

This lab is a great introduction to recursion. The ratio of simple exercises to more challenging ones is perfect, as mastering the structure of easier ones really helps with solving the more difficult problems (i.e. subsets). My only addition to this lab would be a few more quiz questions, i.e. about identifying base cases, etc. 

Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-selftest.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html  
Type of Issue: Could give more info
Severity: minor
Comments: I think it would be useful if there were a question at the end saying something like "how could your answers to the above quiz questions help with making your recursive call or base case?". That might make approaching this problem (which is more challenging than the others) a bit less intimidating. 

Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursive-reporters-intro.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
Type of Issue: Confusing info
Severity: significant
Comments: The last example about recursive calls appearing in the report directly is pretty confusing. I think adding a different example or elaborating on this particular example might clarify things considerably.

**Comment by juliaalcaraz** [comment]
Your Name: Julia Alcaraz

Overview of Lab Comments: This lab does a good job in exposing the students to more challenging recursive problems that no longer use command blocks, as were used for trees. I thought the large amount of exercises was really helpful because it allows the students to get the necessary practice they need to thoroughly understand the way that recursion works. Overall, I liked the lab.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursive-reporters-examples.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&course=cs10_sp15.html&novideo&noreading&noassignment
        Type of Issue: repetitive information
        Severity: minor-ish inconvenience
        Comments: The entire information on this page is also included on the previous page, so it is slightly confusing as to whether it really is the second page. I would suggest removing it from the first page and just leaving this second page, so that the first page is not as dense, and the student can read two slightly less dense pages of information.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-selftest.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&course=cs10_sp15.html&novideo&noreading&noassignment
        Type of Issue: suggestion
        Severity: minor inconvenience
        Comments: Here maybe add a warning for the students to remain on this page until they have attempted coding Pascal's triangle, so that they do not accidentally move on to the next page and find the solution.

**Comment by susanshen96** [comment]
Your Name: Susan Shen

Overview of Lab Comments: I think this lab, once I figured out how the scripts worked and how to visualize recursion, really helped me establish a conceptual basis for recursion that still helps me today, when I solve recursive problems in CS 61B. I think TAs should really emphasize how important it is to understand recursion because it can make some iterative problems much easier and more simplified. 

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursive-reporters-examples.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html 
        Type of Issue: Redundancy 
        Severity: minor inconvenience
        Comments: It seems that the four sample scripts that appeared on the previous slide have reappeared, and I feel it’s a little redundant to talk about these scripts again when the previous slide already asked for the students to understand how they work. 
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-selftest.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: Difficulty for student
        Severity: i.e. Minor inconvenience 
        Comments: In question 1, how the Pascal’s Triangle is set up might be difficult for the student to find out which numbers sum up to the next number on the triangle. It might be easier for the student to solve the question if the triangle were formatted in the same manner as before, with neat rows and lines.

**Comment by Meghna15** [comment]
Your Name: Meghna Dasgupta

Overview of Lab Comments: I think this lab is a very effective way to introduce recursive reporters to the students as it provides enough examples for them to practice with. One thing that I noticed was that there was only one line about mutual recursion added at the bottom of the second last page so it may be more helpful to explain it a little more in detail. Overall the lab is well-structured and fairly paced. 
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursive-reporters-intro.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html      
    Type of Issue: Can be made easier 
        Severity: Minor
        Comments: It would help to put the different combiners into tables so that it is easier to process.

**Comment by sayabe** [comment]
Your Name: Satoko Ayabe

Overview of Lab Comments: I liked how there were messages at beginning of the lab/throughout the lab like "it's highly recommended that you ask for help"/"don't go on until you understand". It helps the students feel at ease because it's hard to get a hang of recursion at first. I also like how they give examples of how to combine answers (e.g. by using operators, or by using join __ __ )

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-and-efficiency.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: i.e. lack of clarity
        Severity: i.e. minor inconvenience
        Comments: To be honest, it might be better to show the solution of the apple/banana subsets code, and ask what it would do/output instead of having students figure it out. I remember struggling on this for such a long time during lab last semester.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursively-sorting-merge.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: i.e. lack of clarity
        Severity: i.e. minor inconvenience
        Comments: The page includes the 2 questions: "Finish this definition by filling the else slot.", "Do you have to check for the two first items being equal? Why or why not?". I think it'll be helpful if the page also provides answers to these questions ("click to show/hide answer"). I think it could save a lot of time because TAs/Lab assistants won't have to go around helping everyone with the 2 questions.

**Comment by Justin-Kim72** [comment]
Name: Justin Kim
Overview of the lab comments: I think the lab is excellent. The only issue is that I don't see why there are answers to some of the questions. I can see how that is benefit, allowing them to see the code and then interpret it from there, but I think it's more important for them to come up with the idea. I am afraid that formatting it the way that it is in the lab will make kids just kind of glance over the answers, instead of really delving into the coding. I just feel like making the students think and come up with the coding is more beneficial to them instead of showing the answers in the next slide. My opinion, but other than that, the lab will be solid and new challenging concept for the students.

**Comment by thorsoch** [comment]
Your Name: Charles Thorson

Overview of Lab Comments: Recursion has always been a fun concept to meddle with, because it almost feels like you are cheating when the program finds your answer. Overall, the lab is good. The exercises are thorough and the length seems just right considering how confusing the concept seems at first. Excellent explanations in the lab.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-selftest.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: bug
        Severity: inconvenience, can cause confusion if student does not notice
        Comments: Question 2, the comments that appear next to the question are correct, but it mistakenly identifies the third choice as the correct answer.

**Comment by 18epedersen** [comment]
Your Name: Emily Pedersen 

Overview of Lab Comments: I really like how the first page introduces the general formula for a recursive reporter. However, I think we should stress even more that the recursive case reports a "combiner" and the function with a smaller input even more because when I was going through the problems as a CS10 student I still didn't quite grasp that concept. I also like how the lab introduces sorting and merging lists because one of the CS61A labs involves solving those kinds of problems. 
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/plurals-of-words.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: lack of clarity
        Severity: prevents students from continuing 
        Comments: I think to guide the students in doing the exaggerate block, we should also post the iterative solution. I think the exaggerate block is a bit too difficult without such guidance and students may get stuck and not be able to continue with the lab. 
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-of-a-set.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: lack of clarity
        Severity: prevents students from continuing 
        Comments: I found this page and the subsequent pages regarding subsets really confusing. What is the student suppose to do? Is she supposed to make the block or simply understand the idea?

**Comment by Stantlers** [comment]
Your Name: Stanley Ho

Overview of Lab Comments: I like how the general formulas for recursive reporters are discussed in the beginning. I also really like how the answer to the blocks are given and carefully explained before you can see the answer.

**Comment by megancarey** [comment]
Your Name: Megan Carey

Overview of Lab Comments: a great, effective lab! I think that while the problems are a little difficult to sort out the first time around, the example problems in the text provide good conceptual templates to work them out.

      Just a note -- throughout the earlier labs, there would be reminder pages to have students switch off, allowing both students time a the computer. I think that since there are so many example problems in this lab it would be especially important that both students get practice working them out.

**Comment by keehansullivan** [comment]
Your Name: Sean Sullivan

Overview of Lab Comments: Lab was definitely great for recursion! I do not personally remember the subsets portion of the lab when I took the course, and think it was unnecessarily confusing for trying to teach recursion, however, if it is used to also introduce another topic then I am all for it. As stated in a previous lab review for the same lab, completing the pascals triangle block before moving on is great for practice, and I felt a little discouraged when I saw the answer on the next page. Also, explicitly mentioning the more base cases you have for a recursive problem usually makes the block more efficient. I believe it was briefly said when counting the time it took the given pascals triangle one, but I think it is an important side note and deserves a sentence or two.

**Comment by morganewing** [comment]
Your Name: Morgan Ewing

Overview of Lab Comments: Overall, I think this lab is challenging yet manageable. I think one of the most helpful parts of this lab is the first page because it gives the students a template and multiple examples which are similar to the recursion problems they see in the rest of the lab so they are able to go back and reference this page. I think the most difficult concept is about the subsets but the finished block is provided so students shouldn't be stuck for too long. I also like how the students can think through how to build some blocks by answering specific questions.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursively-sorting-merge.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: code bug
        Severity: minor inconvenience
        Comments: The merge block in the starter file does not have the list1 variable in the base case.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-
  reporters/subsets-of-a-set.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: lack of clarity
        Severity: minor inconvenience
        Comments: I think the use of the word "meaningfully" in this sentence: "A list is an ordered sequence—you can meaningfully ask "what is the third item of this list?" But we'll agree that when using a list to represent a set, order doesn't count." makes it a little confusing.

**Comment by eurieoh** [comment]
Your Name: Eurie Oh

Overview of Lab Comments: I think that this lab was a good way to further help students understand recursion. I liked the examples used because many of them are about making blocks they already know how to use with HOFs. This makes the problem more understandable and also forces them to think of recursion as another way to solve the same problem.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/even-number-filter-without-generalization.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
  Type of Issue: Too much information
  Severity: Moderate
  Comments: I think that instead of giving the code for the students to copy, it may be a better idea for them to figure it out themselves.

**Comment by ilinabg** [comment]
Your Name: Ilina Bhaya-Grossman

Overview of Lab Comments: I think that the beginning of this lab is extremely important to cement student understanding of recursion and maybe it could be a little clearer? However, I think that the examples that are laid out will really help with comprehension. I also think that the self test multiple choice before Pascal's triangle is really helpful and for future labs there should be brief self tests before challenging questions (I remember having a lot of trouble understanding what Pascal's triangle set up was last semester).
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursively-sorting-even-odd.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: Clarity
        Severity: minor inconvenience
        Comments: This slide mentions mutual recursion very briefly but I think maybe that should be a topic that is a little more explained? I feel like mutual recursion is tricky to understand and should be described a bit more in this slide.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-of-a-set.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: i.e. lack of clarity,
        Severity: i.e. moderate
        Comments: The definition of subset is really confusing and awkwardly worded.

**Comment by papratt** [comment]
Your Name: Paige Pratt

Overview of Lab Comments: I think this lab is pretty good. Recursion is a tough concept and if students actually try to answer every question they should get a pretty good idea of how to use recursion. I also like how future slides help explain and give hints to the exercises that preceded them. Also I feel like this lab covers most of not all the cases of recursive reporters students will encounter in CS10. I didn't have any specific problems with this lab.

**Comment by carolinekim** [comment]
Your Name: Caroline Kim

Overview of Lab Comments: The lab definitely gives a very thorough explanation and demonstration of different ways students can use recursion in snap. The only concern is that too much is given, and there could be more problems that are given to the students for practice. A lot of information seems to be given in this single lab, so making sure that the students understand the material (during lab check offs and discussion) is going to be important. 

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: Unnecessary parts
        Severity: May prevent students from continuing
        Comments: The explanation of Pascal's triangle in terms of binomial expansion seems unnecessary (especially because the lab even states that the students can skip the section). I think it might just confuse some students with words like binomial expansion and combinatorics along with complicated equations being thrown around. It might just be easiest to get rid of the last section on this lab page. 
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-and-efficiency.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: The answer is given
        Severity: Minor inconvenience
        Comments: The answer to finding the subsets is given without really giving the students a chance to try it themselves. I think the lab should tell the students to try the problem and let them know that the answer is on the following pages so that the students can practice writing the code themselves and consult the answer later.

**Comment by KamikazeTomato** [comment]
Your Name: Henry Chung

Overview of Lab Comments: 
The lab seems extremely dense. Some examples to convey the topics of recursion at hand seem to difficult to understand in and of themselves. To have examples using Pascal's Triangle and MergeSort in a single lab seems like it would be daunting an quite possibly overwhelming to students as they have to familiarize themselves with both the recursive ideas and the added complexity of the examples themselves. It strikes me as a bit counterproductive as these examples may reduce comprehension rather than aiding it.

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursive-reporters-intro.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        
  Type of Issue: Lack of Clarity
  Severity: Minor Inconvenience
  Comments:

While the example of the "recursive reporter" blocks serve as a great example of illustrating the small differences between "recursive command" blocks and "recursive reporter blocks" and the example of the factorial block serves to further illustrate this, I feel as if students would benefit with interacting directly with Snap to build the factorial block themselves rather than simply seeing it in its completed form. 
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/plurals-of-words.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html

      Type of Issue: Lack of Clarity
      Severity: Minor Inconvenience
      Comments: In the case that a student is struggling with concepts basic to Recursion from the last lab, it would be helpful here to take an extra block of text under the "words (plural)" block to explain: 
(1) how the recursive call shortens the list (all but first of___) 
(2) joins the items of the list with a combiner 
(3) how the base case works.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-how-long-does-the-computation-take.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: Clarity
        Severity: "minor inconvenience"
        Comments: When including the "count" incrementing to show students how many times the recursive function is getting called, the student is told to "Modify your pascal script to keep count of recursive calls". The wording suggests that the student is supposed to figure it out, but then the answer is provided immediately. If the answer is to simply be given, some explanation as to WHY the change incrementation happens in the beginning of the function, separate from the base case and the else case in order to make sure that every instance of the call is recorded would be useful.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/even-number-filter-without-generalization.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: lack of clarity
        Severity: i.e. minor inconvenience
        Comments: In the example with the "even numbers" block, the lab states that this case is different as it offers 3 distinct cases (1 base case, and 2 recursive cases) instead of the accustomed two. I feel as if could be much more clearly presented with a bullet point for each possible case and the reasoning for each under each bullet point, instead of one solid mass of text.

**Comment by JaysanM** [comment]
Your Name: Jaysan Maolinbay

Overview of Lab Comments: Overall, this lab would be extremely difficult to complete without a partner. There is a lot packed into one lab. In my belief, there should be more simple, given examples so students can build to the more difficult examples. Without a solid base understanding of the lab, students won't have much direction when it comes to the more difficult problems. However, some students may find the challenge rewarding and may come out with more in the lab because of the difficulty. Overall, this lab is kind of difficult, maybe some more elaboration or examples should be given to the students before the really difficult problems. I imagine this lab will be rewarding when finished, however.
--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-of-a-set.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: Difficulty
        Severity: Minor-Little
        Comments: There is no technical problem with this link. However, I do believe here many students will get stuck because the problem is a little too difficult for the amount of information learned beforehand. I recommend  moving this problem to the end of the lab, or maybe having this replaced with something that is more manageable.

**Comment by xmhjch** [comment]
Your Name: Jiachen Hu

Overview of Lab Comments: This is a pretty hard lab especially with only one hour and a half to finish, even with a partner. The explanations and instructions for recursive reporter are reasonable, but the exercise is definitely more difficult, for it involved contents from before that they might forgot already. The second part of the fib is pretty clear, since all necessary codes are provided it would not be so bad. The third part however, for merge list. It would be really challenging for the students especially since they had nearly no knowledge of sorting means. The subset part is also tricky. I think it would be helpful is some helper selection of code or pseudo code is provided.

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link:http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursively-sorting-merge.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html

      Type of Issue: lack of clarity
      Severity: minor
      Comments: It would be much more helpful if there are animations or gifs illustrating how merge sort work. Since merge sort first appear in cs61b, it might be a bit hard for them to understand the scheme of how merge sort work.

**Comment by amy97** [comment]
Name: Amy Vatcha

Overview of Lab Comments: Great lab, integrates important concepts of computer science while giving students many opportunities to practice making blocks with the concepts learnt throughout the lab.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: Llack of clarity
        Severity: Minor inconvenience
        Comments: In the last line, defining combinatorics would be helpful and would improve clarity.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-of-a-set-fast-soln.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: Incomplete
        Severity: Minor inconvenience
        Comments: The lab finishes very abruptly. In my opinion, the last page should sum up the concepts learnt and how it relates to the previous and/or future labs. For example, comparing the concepts learnt in this lab to those learnt in the previous recursion lab would be helpful.

**Comment by lizapraprotnik** [comment]
This is a challenging, yet very informative, lab. The mergesort and subset parts of the lab are pretty challenging, and I think that if more basic examples were included before introducing these two problems, the students would be more prepared to solve them. I also think it would be helpful to remind students to think about the runtime of their code, since that topic was already discussed in a previous lab.

Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursively-sorting-even-odd.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html 
Type of Issue: lack of clarity in instrucitons
Severity: minor
Comments: I think that this page should encourage students to try to create the sort block on their own before giving them the answer on the next page.

**Comment by frankyguerrero** [comment]
Your Name: Rocio Guerrero

Overview of Lab Comments: This lab was very informative and helpful regarding recursion. Although there are quite a number of exercises, I think the exercises are of a good difficulty level in preparation for the midterm and the final. I also think it's actually good to put a lot of exercises because not all of them are required for lab check-offs but are good practice for the students to fully understand recursion.

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/%7Ecs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: lack of clarity
        Severity: minor inconvenience
        Comments: seems like a pretty confusing concept at first. I'm certain a lot of the students will ask questions about this. But solutions are found in the next pages so those will be very helpful in explaining the concept.

**Comment by cynthiaha** [comment]
Your Name: Cynthia Ha

Overview of Lab Comments: 
Overall I think this lab was great and very informative. Although challenging, students are able to really practice executing recursive functions and hence reinforce the ideas they learn in lecture. The pascal's triangle and merge functions are tricky and it is definitely useful to encourage students to work in pairs.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursive-reporters-intro.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html

      Type of Issue: Lack of Clarity
      Severity: minor inconvenience
      Comments: There is a lot of code on the screen, which may be difficult for students to understand on one page. The examples, especially the last one, may not be clear to some students and so it may be better to include a short description of what the block does (or providing example inputs and outputs). Furthermore, since students are advised not to proceed until they understand how the 4 scripts work, it may be useful to add a few multiple choice questions to test their understanding.

**Comment by danielduazo** [comment]
Your Name: Daniel Duazo

Overview of Lab Comments: I found this lab to be one of the most difficult out of all of the CS10 (also happens to be one of my favorites). I think it is a lot of information to digest from some students. For others, it may seem too long for the amount of lab time allotted.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-selftest.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&course=cs10_sp15.html&noassignment&noreading&novideo
        Type of Issue: formatting
        Severity: minor inconvenience
        Comments: For some who may not be comfortable with pascal's triangle, the formatting of the question (specifically, pascal's triangle looking more like a right triangle) makes it really tricky and confusing to understand (especially when looking at rows, columns).

**Comment by benlsmith** [comment]
Your Name: Benjamin Smith

**Comment by benlsmith** [comment]
Your Name: Benjamin Smith
Overview of Lab Comments: This lab is extremely helpful for building upon recursion however seemed really dense for some parts, and was fairly challenging. The subsets portion was pretty difficult, and seemed like a bit much to ask given the time that the students have. Also, i felt that for the more difficult sections, a bit more helper code would be convenient and help cut down a lot of busy time, while still helping them grasp the concepts of recursion.

**Comment by soham14** [comment]
Your Name: Soham Kudtarkar

Overview of Lab Comments: This was a very helpful but very dense lab. I feel that although recursive reporters are a very heavy and very important topic, the lab could have included less challenging and more intuitive problems. I found that the fact that multiple base cases could occur in a recursive reporter problem was not as emphasized as it could have been. I felt that there was a lot of work, both conceptual and practical, for some of the problems that sometimes wasn't necessary to understanding the fundamentals of recursive reporters.

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-of-a-set.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: lack of clarity
        Severity: i.e. minor inconvenience
        Comments: I don't believe that the subsets problem is one that can be easily solved recursively for students who are just starting out with recursion. I feel that more guidance should be given here.

**Comment by JiazhenChen** [comment]
Your Name: Jiazhen Chen

Overview of Lab Comments: This lab is a really nice one. Recursion is an important and useful idea in computer science and I think this lab gives a very good introduction. The examples are very infromative and are at the right difficulty level. I think the suggestion at the bottom of the first page is really a good one. The best way to understand recursion is indeed writing out the logical process behind by oneself. The examples on the first pages serve as great practices. 

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-self-test.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: clarity
        Severity: minor inconvenience
        Comments: The hint questions for the subset question are actually nice. I've already done the lab before so I can easily understand that Question 3 acutally encourage the students to think about dividing the subsets into those have the word "Apple" and those that don't have. But the students may not be able to understand this hint. Maybe adding one more sentence like "Think about dividng the subsets into two parts." may help a bit more.

**Comment by candidia** [comment]
Your Name: Florin-James Langer

Overview of Lab Comments: I like the part where it's optional to skip the rest if you don't like math. It starts getting really tricky at the end, so I like that you give answers the page after.

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-selftest.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: clarity
        Severity: not severe
        Comments: Question 1 is a bit confusing the way it's worded. I would color coat the "and"s, so that it's easier to tell what's going on.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-how-long-does-the-computation-take.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: missing picture
        Severity: is confusing
        Comments: The bottom image is not working; it just has a question mark.

**Comment by jesslarson** [comment]
Your Name: Jessica Larson

Overview of Lab Comments: I loved this lab! Another one of my favorites!

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: snap code bug
        Type of Issue: code bug
        Severity: prevents student from continuing
        Comments: every time I click on the "_ in front of" it drags over the "item _ of list" block.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursively-sorting-even-odd.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: starter file
        Severity: annoying
        Comments: Give the starter file at the beginning, it is hard to save a lab in one cohesive file if we get starter files towards the end of the lab.

**Comment by rwason** [comment]
Your Name: Rishabh Wason

Overview of Lab Comments: I think that this lab was personally the most challenging lab of the year. If a student fully understands this lab, they will have good control over recursive reporters and will be as competent as this course requires. I think that this lab is definitely worth the length and difficulty it is. I think that it would be nice and convenient if the student could get a starter file of all of the blocks used as examples so that they themselves can try using them on their own. 

Page link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-of-a-set.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html Type of issue: Lack of clarity
Severity: Minor
Comments: I feel like this section could be written in a more concise way. I feel like the concept of a subset as well as its definition is awkwardly worded and it could be written more clearly.

**Comment by seunghalee** [comment]
Your Name: Seungha Lee

Overview of Lab Comments: I think this is a one of the more challenging and dense yet very helpful labs The solutions at the end really helped with the understanding of the material. Having more starter files would have been better and would have definitely saved a lot of time. 

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-how-long-does-the-computation-take.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: Table at the very bottom does not show up
        Severity: i.e. minor inconvenience

**Comment by janicechui** [comment]
Your Name: Janice Chui

Overview of Lab Comments: This lab is generally more challenging than others. The images help the student with the understanding of concepts. 

--- For each specific Issue (Copy this section as many times as necessary) ---

 Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-how-long-does-the-computation-take.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
    Type of Issue: Formula image does not appear on screen       Severity: i.e. minor inconvenience

**Comment by MaazUddin11** [comment]
Your Name: Maaz Uddin

Overview of Lab Comments: Was a good lab, but I feel some people who do not have grasped the concept of recursion properly may find many of the functions to make a bit difficult.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-of-a-set.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: difficulty level
        Severity: prevents students from continuing
        Comments: I remember doing the subset problem last semester and it was very difficult. I feel that the TA should probably give an intro to the whole class on how it works instead of the students doing it themselves entirely. And although the lab does explain how the function works a few pages after, I believe it'll be more of a learning experience if the students had an idea of what to do before they go into it.

**Comment by mikomwang** [comment]
Your Name: Michael Wang

Overview of Lab Comments: Very detailed lab, with many good practice examples. Though it seems  there is a lot of agreement on how dense the lab is, I feel as if it is prepared very well, and it would be best to have students face many practice problems with recursion now so they are prepared for the midterm. Additionally, I see a lot of these examples being repeated in 61a, such as mergesort and pascal's triangle. In my opinion, in order to fully convey the importance and relevancy of this lab, it might be important to relate what they're currently doing in this class to what they'll be doing in future CS classes. Maybe even in the python segment of cs10 we can bring these questions back and ask them to write it recursively in python syntax. 
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-how-long-does-the-computation-take.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: Faulty image
        Severity: Minor
        Comments: The image at the bottom of the screen does not display correctly. 

-[ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
      Type of Issue: Confusing
      Severity: i.e. "prevents students from continuing", "minor inconvenience", etc
      Comments: I don't think this math formula is entirely necessary to understand how Pascal's triangle works. In fact, I feel like it may even overwhelm students who are already struggling.

**Comment by msilva926** [comment]
Your Name: Melanie Silva

Overview of Lab Comments: i.e. Was this lab good or bad? Too fast? Too Slow? Does it need something new?
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursive-reporters-intro.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: none
        Severity: none
        Comments: I like how there is an option for students to go beyond the lab and try something challenging. "Optional exercise for hotshots: How did we make that picture?" Some students have a faster pace and this gives them something to do and make them hangout in lab a little longer. There should be more of these optional exercises. 
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/plurals-of-words.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: none
        Severity: none
        Comments: Recursion is one of the most difficult topics, and I like how there are a lot of hints to get students started and thinking about which blocks are good for recursion.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-how-long-does-the-computation-take.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: image issue
        Severity: minor inconvenience
        Comments: There is a broken image at the bottom that is supposed to show the math formula for calculating the pascal triangle values.

**Comment by Laralinmcc** [comment]
Lara McConnaughey

Overview of Lab Comments: i.e. This lab is well paced and has some great visuals for the exercises. There is a lot of lack of clarity in the lab and I think that the structure of the explanations of some of the blocks could be better. The Instructions for some of the exercises could be layer out in a format with bullet points and I think that would fix the lack of clarity issues I found in this lab.

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursive-reporters-intro.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&course=cs10_sp15.html&novideo&noreading&noassignment
        Type of Issue:lack of clarity
        Severity: i.e. "minor inconvenience"
        Comments: This page talks a lot about combining two things using a combining block, but doesn't really go into what you are combining. I think it would be helpful if it was mentioned that things you are combining are the recursive calls of a function. That's a very hard topic to explain, but I think that it should be mentioned somewhere. Also a list of common combining blocks somewhere would be nice.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/plurals-of-words.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&course=cs10_sp15.html&novideo&noreading&noassignment
        Type of Issue: i.e. lack of clarity
        Severity: i.e. "minor inconvenience"
        Comments: for the instructions in the make an exaggerate block it has four hints at the end instead of saying Hint it would be better if the steps or hints were just in bullet points.
- [ ] Page Link:http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/even-number-filter-without-generalization.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&course=cs10_sp15.html&novideo&noreading&noassignment
        Type of Issue: lack of clarity
        Severity: i.e. "minor inconvenience"
        Comments: For the words ending in e exercise the description says that there are two recursive cases. I think somewhere it should specifically state that only one of the recursive cases is called.

**Comment by Rshahatit** [comment]
Your Name: Rami Shahatit

Overview of Lab Comments: I think this lab overall was a good lab on recursion. Being in CS61A I can say that it helped me throughout the class. I even understand recursion better now that I have reviewed it. I know you havent talked about python at all yet in class. but since this is a intro course. I thought maybe you could show them an example of recursion in python that way they have more exposure to python than just the last few weeks.

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursive-reporters-intro.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue:nothing is wrong with it. I remember this page. I like the note at the end that tells students to totally understand all the examples. I think that is important.
        Severity:  A good thing
        Comments: IT is good
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-selftest.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: lack of clarity 
        Severity: "minor inconvenience"
        Comments: I almost overlooked the last little hand sign with the arrow on it so maybe make that bigger or like have a little bit of space between it and the bottom of the page so students see it.
  Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-how-long-does-the-computation-take.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: lack of clarity
        Severity: i.e. "minor inconvenience"
        Comments: I think the picture at the end of this page is missing. It might just be my computer but there is definitely supposed to be a picture there but there is not one.

**Comment by chnicoloso** [comment]
Your Name: Christian Lista-Nicoloso

Overview of Lab Comments: i.e. This lab is overwhelming. However, it does a good job at making sure the student knows that they 'should' feel overwhelmed by this stuff. In the very beginning it says that it'll be challenging, then when talking about Pascal Triangles it tells you that you can skip the deep mathematical stuff, etc. That being said, I wish the lab made it seem more clear or obvious that most recursive functions will be almost identical. Yes, it does mention that it they all have a base case and a recursive case but I would like for the lab to specifically point out the huge similarities between all of the functions that the students will build in the lab. I remember that when I took the class, I thought every recursive function would have something crazy or special about them, and some of them do, but for the most part, they are all the same. I really like the Hints in the lab. They are very helpful and offer very good guidance.

**Comment by KimiaG** [comment]
Kimia Ghodoosian

Overview of Lab Comments: i.e. Was this lab good or bad? Too fast? Too Slow? Does it need something new?
Overall, this lab was good. The multiple choice questions were helpful in hinting at how to implement a block. I think it might be helpful to have one tree-diagram representation somewhere in the lab (maybe interactive!?) because I remember seeing that made recursion seen less like "magic" to me and helped me understand the process behind it. For a lot of the exercises, the answer was given on the next slide. It's true that this is a challenging lab, but I think some more hints rather than the answer might be more helpful and along the way students would hopefully gain a better understanding of what is happening at each step of the recursive process. 
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/recursively-sorting-merge.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&course=cs10_sp15.html&novideo&noreading&noassignment
        Type of Issue: lack of clarity
        Severity: confusing, may prevent students from continuing
        Comments: I remember having a lot of difficult with this page. The merge block is given with little to no explanation, and though it may seem self-explanatory I remember being really confused when I first saw it and the Lab Assistants were not able to provide a very clear explanation. A brief explanation of what is happening in the recursive call should be enough clarification.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-and-efficiency.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&course=cs10_sp15.html&novideo&noreading&noassignment
  Type of Issue: lack of clarity
        Severity: doesn't test understanding
        Comments: I think the answer here is also given too easily with the explanation given only after the answer is given. As a student rushing to finish a difficult lab, I probably wouldn't read this section, and I would regret it on the midterm. To fix the issue, consider re-ordering so that the explanation precedes the answer, and encourage students to devise a method that they would use to divide a list of objects into its subsets and writing out pseudo code for what they come up with.

**Comment by michen-ml** [comment]
Michelle Chen

Overview of Lab Comments: The lab will probably be pretty difficult for most of the students,  so it's really helpful that there's a lot of examples and practice problems. The mathematical part of the Pascal's triangle might be unnecessary, since most people probably wouldn't read it, and it doesn't really add much to the topic of recursion. Otherwise, it's a well-balanced lab.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/subsets-of-a-set-fast-soln.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: clarity
        Severity: minor
        Comments: Some explanation of how the two blocks differ/why one is faster than the other might help with clarity. It might not be apparent right away, so some explanation would help.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/pascals-triangle-how-long-does-the-computation-take.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: clarity
        Severity: minor
        Comments: The explanation of why redundancy in recursion matters is kind of vague. It doesn't really explain why we should care about redundancy, and mostly just explains how pascal's triangle is redundant.

**Comment by siyingwang** [comment]
Your Name: Siying Wang

Overview of Lab Comments: This is a changeling lab for students, but it is good at explaining recursion. And all the examples shows different important points in the recursion. 

--- For each specific Issue (Copy this section as many times as necessary) ---
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/recursive-reporters/plurals-of-words.html?topic=berkeley_bjc%2Frecur%2Frecursive-reporters-part1.topic&novideo&noreading&noassingment&course=cs10_sp15.html
        Type of Issue: lack of clarity
        Severity: minor inconvenience
        Comments: I found that students constantly have questions on exaggerate. I think it will be better to give a hint for them to use a helper function to deal with the single word and number.

**Comment by benlsmith** [comment]
Your Name: Benjamin Smith
Overview of Lab Comments: This lab did a really good job of exposing students to higher level recursive problems. It did seem to be a bit challenging and time intensive, so taking out some of the exercises may be helpful. They definitely do help, however not all of the exercises are necessary to learn recursion for future classes so trimming them may be helpful.
